core/interfaces.go

```diff
@@ -149,3 +149,11 @@ type DNSResolver interface {
 	LookupCAA(string) ([]*dns.CAA, time.Duration, error)
 	LookupMX(string) ([]string, time.Duration, error)
 }
+
+// PolicyAuthorityDatabase is the public interface to the white/blacklists
+type PolicyAuthorityDatabase interface {
```

Comment:
@jmhodges: You were explaining to me why narrower interfaces, close to their site of use, are nicer for testing (there's less to mock out). I started to explain here, but I'd be duplicating the argument poorly. Can you provide a reference or advice here?